From dd0c31d6b7559bb0dda6283a5d345d537f1d6b42 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@gmail.com>
Date: Tue, 30 Jul 2013 13:17:45 +1000
Subject: [PATCH] rrcrtc: brackets are hard, lets go shopping.

Slaving two outputs on a secondary GPU to a primary GPU testing
picked this up, in that we'd try to resize to the totally the
wrong thing, then as usual segfault in the rotation code.

Signed-off-by: Dave Airlie <airlied@redhat.com>
---
 randr/rrcrtc.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

Index: xorg-server/randr/rrcrtc.c
===================================================================
--- xorg-server.orig/randr/rrcrtc.c	2014-03-05 15:25:13.634526931 +0000
+++ xorg-server/randr/rrcrtc.c	2014-03-05 15:25:22.938653390 +0000
@@ -476,7 +476,7 @@
 
     xorg_list_for_each_entry(slave, &pScreen->output_slave_list, output_head) {
         rrScrPrivPtr    slave_priv = rrGetScrPriv(slave);
-        for (c = 0; c < slave_priv->numCrtcs; c++)
+        for (c = 0; c < slave_priv->numCrtcs; c++) {
             if (slave_priv->crtcs[c] == rr_crtc) {
                 newbox.x1 = x;
                 newbox.x2 = x + w;
@@ -491,8 +491,9 @@
                 newbox.y1 = slave_priv->crtcs[c]->y;
                 newbox.y2 = slave_priv->crtcs[c]->y + slave_priv->crtcs[c]->mode->mode.height;
             }
-        RegionInit(&new_crtc_region, &newbox, 1);
-        RegionUnion(&total_region, &total_region, &new_crtc_region);
+	    RegionInit(&new_crtc_region, &newbox, 1);
+	    RegionUnion(&total_region, &total_region, &new_crtc_region);
+	}
     }
 
     newsize = RegionExtents(&total_region);
